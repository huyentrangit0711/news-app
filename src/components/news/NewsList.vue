<template>
	<div>
		<div
			class="grid grid-cols-3 gap-4"
			:v-show="articles && articles.length > 0"
		>
			<new-item
				v-for="(article, index) in articles"
				:key="index"
				:article="article"
			></new-item>
		</div>
		<div :v-show="loading">
			Loadinbg
		</div>
		test
	</div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import NewItem from './NewItem.vue';
export default {
	name: 'NewsList',
	components: {
		NewItem,
	},
	data() {
		return {
			loading: false,
		};
	},
	computed: {
		...mapGetters('news', { articles: 'getArticles' }),
		...mapGetters(['activeTab']),
	},
	methods: {
		...mapActions('news', {
			getTopHeadline: 'getTopHeadLineArticles',
			getCustomNews: 'getNewsByQuery',
		}),
	},
	mounted() {
		if (this.activeTab === 0) {
			this.getTopHeadline();
		} else {
			const defaultQuery = 'animal&bitcoin&apple&earthquake';
			this.getCustomNews(defaultQuery);
		}
	},
};
</script>
