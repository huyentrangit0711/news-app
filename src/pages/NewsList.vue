<template>
	<div class="top-headline-tab">
		<div>
			<h1>Top Headline News</h1>
			<router-view></router-view>
		</div>
	</div>
</template>

<script>
export default {
	name: 'NewsListPage',
};
</script>
