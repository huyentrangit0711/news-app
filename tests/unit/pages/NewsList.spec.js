import { shallowMount } from '@vue/test-utils';
import NewsList from '@/pages/NewsList';

describe('Profile Page Component', () => {
	// render the component
	let wrapper;
	beforeEach(() => {});
	afterEach(() => {
		jest.clearAllMocks();
	});
	it('Page Profile should render welcome page correctly', () => {
		wrapper = shallowMount(NewsList, {
			stubs: ['router-link', 'router-view'],
		});
		expect(wrapper).toMatchSnapshot();
	});
});
