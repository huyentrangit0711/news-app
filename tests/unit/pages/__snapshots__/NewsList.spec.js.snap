// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page Component Page Profile should render welcome page correctly 1`] = `
<div class="top-headline-tab">
  <div>
    <h1>Top Headline News</h1>
    <router-view-stub></router-view-stub>
  </div>
</div>
`;
